vortex-layout/src/layouts/dict/bloom/sbbf/mod.rs

```diff
@@ -0,0 +1,173 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright the Vortex contributors
+
+mod prune;
+mod tokenizer;
+
+pub use prune::*;
+pub use tokenizer::*;
+use twox_hash::XxHash64;
+use vortex_buffer::{Buffer, BufferMut, ByteBuffer};
+use vortex_error::{VortexResult, vortex_ensure};
+// TODO(aduffy): performance test the twox_hash impl
+
+const XX_SEED: u64 = 0;
+
+// Constants from the Parquet format specification.
+const C0: u32 = 0x47b6137b;
+const C1: u32 = 0x44974d91;
+const C2: u32 = 0x8824ad5b;
+const C3: u32 = 0xa2b7289d;
+const C4: u32 = 0x705495c7;
+const C5: u32 = 0x2df1424b;
+const C6: u32 = 0x9efc4947;
+const C7: u32 = 0x5c6bfb31;
+
+/// The SBBF filter, using the XXH64 hash algorithm with a seed value of 0.
+///
+/// This configuration is identical to what Parquet uses, and the rationale is laid out in
+/// "Split block Bloom filters" <https://arxiv.org/pdf/2101.01719>.
+#[derive(Clone)]
+pub struct Sbbf {
+    blocks: BufferMut<Block>,
+}
+
+impl Sbbf {
+    pub fn new(blocks: BufferMut<Block>) -> Self {
+        Self { blocks }
+    }
+
+    /// Serialize the filter to a new buffer.
+    pub fn serialize(&self) -> ByteBuffer {
+        BufferMut::copy_from(&self.blocks)
+            .freeze()
+            .into_byte_buffer()
+    }
```

Comment:
Could this be just `self.clone().freeze().into_byte_buffer()`? Though it's likely the same thing overall